addons/google_calendar/utils/google_calendar.py

```diff
@@ -71,14 +71,27 @@ def insert(self, values, token=None, timeout=TIMEOUT):
         headers = {'Content-type': 'application/json', 'Authorization': 'Bearer %s' % token}
         if not values.get('id'):
             values['id'] = uuid4().hex
-        self.google_service._do_request(url, json.dumps(values), headers, method='POST', timeout=timeout)
-        return values['id']
+        try:
+            self.google_service._do_request(url, json.dumps(values), headers, method='POST', timeout=timeout)
+        except requests.HTTPError as error:
+            if error.response.status_code != 400:
+                raise error
+            _logger.info("Google event %s can't be created" % values['id'])
+            return error, None
+        return None, values['id']
```

Comment:
This signature change looks unnecessary, and makes the return value weird and inconsistent. Perhaps you could let the exceptions bubble up, and have a generic `try/catch+handle_exception()` in callers that would perform the error differentiation?

```py
# instead of this:
error, google_id = google_service.insert(values, token=token, timeout=timeout)
if error:
    self._google_error_handling(error)

# you'd do this - which gives a simpler, more straightforward API and logic
try:
    google_id = google_service.insert(values, token=token, timeout=timeout)
except HTTPError as e:
    self._google_error_handling(e)
```

Reply:
Indeed, that's better. Should I keep the try/except on patch here
https://github.com/odoo/odoo/blob/8ee4f31201b8550eb80eea897888269d5f4d9f26/addons/google_calendar/utils/google_calendar.py#L87-L93
Or should I move it in google_sync.py to be coherent?

Reply:
> Or should I move it in google_sync.py to be coherent?

I'd move it, so you can keep a consistent error handling logic for `insert` and `patch` :+1:

Reply:
Unfortunately, it don't work anymore. 
```
self.google_service._do_request(...)
```
will call res.raise_for_status().
https://github.com/odoo-dev/odoo/blob/e70840991bc292a9cf175684d050ce064ed3a0d5/addons/google_account/models/google_service.py#L139-L146
It can be avoided if the try/except block is in patch and insert methods of google_calendar.py here:
https://github.com/odoo-dev/odoo/blob/e70840991bc292a9cf175684d050ce064ed3a0d5/addons/google_calendar/utils/google_calendar.py#L69-L75
But if I do the try/except in google_sync.py it seems to be not working anymore. The raise_for_status() raise directly and the exception is not catched :(. Am I missing something here?